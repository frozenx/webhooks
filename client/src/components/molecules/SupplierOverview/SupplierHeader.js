import React, { Fragment } from 'react';
import { FormHeading, TopHeading } from '../../../common/styles';
import Label from '@beans/label';
import { HeaderOverview, StyledList, SupplierOverviewContainer } from './styled';

const SupplierHeader = props => {
  const {
    partnerName,
    staticAttributes,
    values
  } = props;
  return (
    <SupplierOverviewContainer>
      <FormHeading>
        <TopHeading><h1>{partnerName}</h1></TopHeading>
      </FormHeading>
      <HeaderOverview>
        <Label dark={true} emphasized={true} className="supplier-title">
          {' '}
          Partner details
        </Label>
        <StyledList>
          {
            staticAttributes.map((staticAttribute, i) => {
              return (
                <li key={`partner-static-attribute-${i}`}>
                  <Label dark={true} emphasized={true}>
                    {staticAttribute.displayName}
                  </Label>
                  <Label>{values[staticAttribute.key]}</Label>
                </li>
              )
            })
          }
        </StyledList>
      </HeaderOverview>
    </SupplierOverviewContainer>
  );
};

export default SupplierHeader;
