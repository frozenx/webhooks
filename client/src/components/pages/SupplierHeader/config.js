export const presets = {
    status: {
        depFields: ["partnerEndDateActive"],
        depPresets: {
            partnerEndDateActive: {
                depKey: 'showToggle',
                val: 'Inactive',
                op: '===',
                return: true,
                default: false
            }
        }
    },
    isPrimaryAddFlag: {
        depFields: ['addressName'],
        depPresets: {
            addressName: {
                depKey: 'required',
                val: 'Yes',
                op: '===',
                return: true,
                default: false
            }
        }
    },
    uuid: {
        depFields: ['expiredDate'],
        depPresets: {
            expiredDate: {
                depKey: 'required',
                val: 'uuid',
                op: '===',
                return: false,
                default: true
            }
        }
    },
    contactStatus: {
        depFields: ['inactiveDate'],
        depPresets: {
            inactiveDate: [{
                depKey: 'required',
                val: 'A',
                op: '===',
                return: false,
                default: true
            }, {
                depKey: 'regexRule',
                val: 'A',
                op: '===',
                return: "",
                default: ""
            }],
            administrativeContactFlag: {
                depKey: 'required',
                val: 'A',
                op: '===',
                return: false,
                default: true
            }
        }
    }
}