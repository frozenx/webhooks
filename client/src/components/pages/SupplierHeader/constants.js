const contactResponse = [
    {
        "uuid": "1234",
        "partnerHeaderUuid": "1500d6d5-3cd5-3703-9307-26531c14db0a",
        "addressUuid": "c3e74a2f-2fc9-450b-b321-4319120b9f23",
        "expiryDate": null,
        "is_active": null,
        "firstName": "First Name",
        "lastName": "Last Name",
        "role": "Correspondence contact",
        "location": "Manchester",
        "expiredDate": "03-07-2019",
        "isActive": "Active"
    },
    {
        "uuid": "2345",
        "partnerHeaderUuid": "1500d6d5-3cd5-3703-9307-26531c14db0a",
        "addressUuid": "018100e1-1004-4148-94a1-bd120ac3d679",
        "expiryDate": null,
        "is_active": null,
        "firstName": "First Name",
        "lastName": "Last Name",
        "role": "Administrative contact",
        "location": "Liverpool",
        "expiredDate": "03-07-2019",
        "isActive": "Inactive"
    },
    {
        "uuid": "2345",
        "partnerHeaderUuid": "1500d6d5-3cd5-3703-9307-26531c14db0a",
        "addressUuid": "018100e1-1004-4148-94a1-bd120ac3d679",
        "expiryDate": null,
        "is_active": null,
        "firstName": "First Name",
        "lastName": "Last Name",
        "role": "Ship to contact",
        "location": "Manchester",
        "expiredDate": "03-07-2019",
        "isActive": "Active"
    }
]
export default contactResponse;