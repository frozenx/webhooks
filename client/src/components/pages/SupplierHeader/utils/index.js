import { Map } from 'immutable';


export const updateAttributeRules = (presets, attributeToRulesMapping, attributeKey, attributeValue, entityType) => {
    const updatedRulesMapping = Map(attributeToRulesMapping);

    const presetObject = presets[attributeKey];
    if (presetObject) {
        presetObject.depFields.forEach((depField) => {
            const attributeObject = updatedRulesMapping.get(depField);
            const presetField = presetObject.depPresets[depField];
            const presetFieldEntityType = presetObject.depPresets.entityType
            if (Array.isArray(presetField)) {
                presetField.forEach((field) => {
                    updateAttributeRule(field, attributeValue, attributeObject, entityType, presetFieldEntityType)
                })
            } else {
                updateAttributeRule(presetField, attributeValue, attributeObject, entityType, presetFieldEntityType)
            }
        });
    }
    return updatedRulesMapping.toJS();
}

function regexCheck(a, b) {
  return b.test(a);
}

function equalityCheck(a, b) {
  return b.split(/[ ]*,[ ]*/).indexOf(a) !== -1;
}

function updateAttributeRule(
  attribute,
  attributeValue,
  attributeObject,
  entityType,
  presetFieldEntityType,
) {
  // function that performs a side effect - mutates object attributeObject
  if (entityType === presetFieldEntityType) {
    const operation = attribute.op;
    let presetFunction = equalityCheck;
    if (operation === 'match') {
      presetFunction = regexCheck;
    }
    if (presetFunction(attributeValue, attribute.val)) {
      attributeObject[attribute.depKey] =
        attribute.return[attributeValue] || attribute.return;
    } else {
      attributeObject[attribute.depKey] = attribute.default;
    }
  }
}
