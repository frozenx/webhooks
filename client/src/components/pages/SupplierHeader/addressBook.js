import { pageSize } from "../../../config";

class AddressBook {
    constructor(entries = []) {
        this.getAddressEntries = this.getAddressEntries.bind(this);
        this.editAddress = this.editAddress.bind(this);
        this.addAddress = this.addAddress.bind(this);
        this.linkAddress = this.linkAddress.bind(this);
        this.getAddress = this.getAddress.bind(this);
        this.entries = entries;
        this.total = this.entries.length;
        this.pageSize = pageSize
        this.offset = Math.floor(this.total / this.pageSize);
    }

    getAddressEntries() {
        return this.entries;
    }

    getAddress(id) {
        let address = this.entries.find(address => address.addressUuid === id)
        if (address) {
            return address;
        }
        return null;
    }

    editAddress(id, editedAddress) {
        let foundEntry = this.entries.find(address => address.addressUuid === id)
        if (foundEntry) {
            let foundIndex = this.entries.findIndex(entry => entry.addressUuid == id)
            return this.entries[foundIndex] = editedAddress
        }
    }

    addAddress(address) {
        this.entries.push(address)
    }

    linkAddress(siteId, linkAddressToSite) {
        this.entries.map((address, i) => {
            linkAddressToSite.map((addressId, j) => {
                if(address.addressUuid == addressId) {
                    this.entries[i]['sites'].push(siteId)
                }
            })
        })
    }
    addEntries(entries, addressCount, index) {
        this.entries = entries;
        this.currentPage = index;
        this.total = addressCount;

    }
}

export default AddressBook;