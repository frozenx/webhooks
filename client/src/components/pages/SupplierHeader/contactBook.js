import { pageSize } from "../../../config";

class ContactBook {
    constructor(entries = []) {
        this.getContactEntries = this.getContactEntries.bind(this);
        this.editContact = this.editContact.bind(this);
        this.addContact = this.addContact.bind(this);
        this.linkContact = this.linkContact.bind(this);
        this.entries = entries;
        this.total = this.entries.length;
        this.pageSize = pageSize
        this.offset = Math.floor(this.total/this.pageSize);  
    }

    getContactEntries() {
        return this.entries;
    }

    editContact(id, editedContact) {
        let foundEntry = this.entries.find(contact => contact.id === id)
        if (foundEntry) {
            let foundIndex = this.entries.findIndex(entry => entry.id == id)
            return this.entries[foundIndex] = editedContact
        }           
    }

    getContact(id) {
        let contact = this.entries.find(contact => contact.contactUuid === id)
        if (contact) {
            return contact;
        }
        return null;
    }

    addContact(contact) {
        this.entries.push(contact)
    }

    linkContact (siteId, linkContactsToSite) {
        this.entries.map((contact, i) => {
            linkContactsToSite.map((contactId, j) => {
                if(contact.contactUuid == contactId) {
                    this.entries[i]['sites'].push(siteId)
                }
            })
        })
    }
    
    addEntries(entries, contactCount, index) {
        this.entries = entries;
        this.currentPage = index;
        this.total = contactCount;

    }
}

export default ContactBook;