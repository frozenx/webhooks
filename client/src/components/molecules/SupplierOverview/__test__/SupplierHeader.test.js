import React from "react";
import { shallow } from "enzyme";
import SupplierHeader from "../SupplierHeader";

describe("Supplier header overview card component", () => {
  let Wrapper, partnerAttributes;
  const supplierHeaderData = {
   partnerAttributes : [
    {
      attributeGroup: "partner",
      attributes: [
        {name: "partnerName", values: [{value: "Uniliver", lang: "en"}]},
        {name: "CompanyHouseName", values: [{value: "uni", lang: "en"}]},
        {name: "CompanyRegistrstionNumber", values: [{value: "1", lang: "en"}]},
        {name: "VatNumber", values: [{value: "1", lang: "en"}]},
      ]
    }
  ]
}

const partnerName = "Uniliver"

  beforeEach(() => {
    Wrapper = shallow(<SupplierHeader supplierHeaderData={supplierHeaderData} partnerName={partnerName}/>);
  });

  it("renders the component", () => {
    expect(Wrapper).toMatchSnapshot();
  });
});
