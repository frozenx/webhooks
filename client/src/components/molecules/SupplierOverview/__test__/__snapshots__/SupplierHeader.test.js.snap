// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Supplier header overview card component renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SupplierHeader
    partnerName="Uniliver"
    supplierHeaderData={
      Object {
        "partnerAttributes": Array [
          Object {
            "attributeGroup": "partner",
            "attributes": Array [
              Object {
                "name": "partnerName",
                "values": Array [
                  Object {
                    "lang": "en",
                    "value": "Uniliver",
                  },
                ],
              },
              Object {
                "name": "CompanyHouseName",
                "values": Array [
                  Object {
                    "lang": "en",
                    "value": "uni",
                  },
                ],
              },
              Object {
                "name": "CompanyRegistrstionNumber",
                "values": Array [
                  Object {
                    "lang": "en",
                    "value": "1",
                  },
                ],
              },
              Object {
                "name": "VatNumber",
                "values": Array [
                  Object {
                    "lang": "en",
                    "value": "1",
                  },
                ],
              },
            ],
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          <h1>
            Uniliver
          </h1>
        </ForwardRef>,
        <ForwardRef>
          <_default
            className="supplier-title"
            dark={true}
            emphasized={true}
          >
             Supplier details
          </_default>
          <ul>
            <li>
              <_default
                dark={true}
                emphasized={true}
              >
                partnerName
              </_default>
              <_default>
                Uniliver
              </_default>
            </li>
            <li>
              <_default
                dark={true}
                emphasized={true}
              >
                CompanyHouseName
              </_default>
              <_default>
                uni
              </_default>
            </li>
            <li>
              <_default
                dark={true}
                emphasized={true}
              >
                CompanyRegistrstionNumber
              </_default>
              <_default>
                1
              </_default>
            </li>
            <li>
              <_default
                dark={true}
                emphasized={true}
              >
                VatNumber
              </_default>
              <_default>
                1
              </_default>
            </li>
          </ul>
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h1>
            Uniliver
          </h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Uniliver",
          },
          "ref": null,
          "rendered": "Uniliver",
          "type": "h1",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": false,
            "rules": Array [
              [Function],
              ";",
              "-webkit-tap-highlight-color:",
              [Function],
              ";border:0;box-sizing:border-box;margin:0;outline:",
              [Function],
              ";padding:0;position:relative;",
              ";",
              "
    margin-top: 34px;
    .input-error {
        color: #cc3333;
    }
",
            ],
          },
          "displayName": "Styled(base-components__RootElement)",
          "foldedComponentIds": Array [
            "base-components__RootElement-sc-150tvch-1",
          ],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <_default
              className="supplier-title"
              dark={true}
              emphasized={true}
            >
               Supplier details
            </_default>,
            <ul>
              <li>
                <_default
                  dark={true}
                  emphasized={true}
                >
                  partnerName
                </_default>
                <_default>
                  Uniliver
                </_default>
              </li>
              <li>
                <_default
                  dark={true}
                  emphasized={true}
                >
                  CompanyHouseName
                </_default>
                <_default>
                  uni
                </_default>
              </li>
              <li>
                <_default
                  dark={true}
                  emphasized={true}
                >
                  CompanyRegistrstionNumber
                </_default>
                <_default>
                  1
                </_default>
              </li>
              <li>
                <_default
                  dark={true}
                  emphasized={true}
                >
                  VatNumber
                </_default>
                <_default>
                  1
                </_default>
              </li>
            </ul>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": " Supplier details",
              "className": "supplier-title",
              "dark": true,
              "emphasized": true,
            },
            "ref": null,
            "rendered": " Supplier details",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  <li>
                    <_default
                      dark={true}
                      emphasized={true}
                    >
                      partnerName
                    </_default>
                    <_default>
                      Uniliver
                    </_default>
                  </li>,
                  <li>
                    <_default
                      dark={true}
                      emphasized={true}
                    >
                      CompanyHouseName
                    </_default>
                    <_default>
                      uni
                    </_default>
                  </li>,
                  <li>
                    <_default
                      dark={true}
                      emphasized={true}
                    >
                      CompanyRegistrstionNumber
                    </_default>
                    <_default>
                      1
                    </_default>
                  </li>,
                  <li>
                    <_default
                      dark={true}
                      emphasized={true}
                    >
                      VatNumber
                    </_default>
                    <_default>
                      1
                    </_default>
                  </li>,
                ],
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <_default
                      dark={true}
                      emphasized={true}
                    >
                      partnerName
                    </_default>,
                    <_default>
                      Uniliver
                    </_default>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "partnerName",
                      "dark": true,
                      "emphasized": true,
                    },
                    "ref": null,
                    "rendered": "partnerName",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Uniliver",
                    },
                    "ref": null,
                    "rendered": "Uniliver",
                    "type": [Function],
                  },
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <_default
                      dark={true}
                      emphasized={true}
                    >
                      CompanyHouseName
                    </_default>,
                    <_default>
                      uni
                    </_default>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "CompanyHouseName",
                      "dark": true,
                      "emphasized": true,
                    },
                    "ref": null,
                    "rendered": "CompanyHouseName",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "uni",
                    },
                    "ref": null,
                    "rendered": "uni",
                    "type": [Function],
                  },
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <_default
                      dark={true}
                      emphasized={true}
                    >
                      CompanyRegistrstionNumber
                    </_default>,
                    <_default>
                      1
                    </_default>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "CompanyRegistrstionNumber",
                      "dark": true,
                      "emphasized": true,
                    },
                    "ref": null,
                    "rendered": "CompanyRegistrstionNumber",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "1",
                    },
                    "ref": null,
                    "rendered": "1",
                    "type": [Function],
                  },
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <_default
                      dark={true}
                      emphasized={true}
                    >
                      VatNumber
                    </_default>,
                    <_default>
                      1
                    </_default>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "VatNumber",
                      "dark": true,
                      "emphasized": true,
                    },
                    "ref": null,
                    "rendered": "VatNumber",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "1",
                    },
                    "ref": null,
                    "rendered": "1",
                    "type": [Function],
                  },
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-VigVT",
            "isStatic": true,
            "rules": Array [
              "
    background-color: #f6f6f6;
    padding: 28px 12px 16px 12px;
    border: solid 1px #e5e5e5;
    label.supplier-title{
        font-size: 20px;
        margin-bottom: 20px
    }
    ul{
        margin:0
        padding:0
        li {
            width: 25%;
            display: inline-block;
            margin-bottom: 20px;
            label{
                width: 100%;
            }                  
            @media (max-width: 756px) {
                width: 50%;
            }  
            @media (max-width: 505px) {
                width: 100%;
            }       
        }
    }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-VigVT",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            <h1>
              Uniliver
            </h1>
          </ForwardRef>,
          <ForwardRef>
            <_default
              className="supplier-title"
              dark={true}
              emphasized={true}
            >
               Supplier details
            </_default>
            <ul>
              <li>
                <_default
                  dark={true}
                  emphasized={true}
                >
                  partnerName
                </_default>
                <_default>
                  Uniliver
                </_default>
              </li>
              <li>
                <_default
                  dark={true}
                  emphasized={true}
                >
                  CompanyHouseName
                </_default>
                <_default>
                  uni
                </_default>
              </li>
              <li>
                <_default
                  dark={true}
                  emphasized={true}
                >
                  CompanyRegistrstionNumber
                </_default>
                <_default>
                  1
                </_default>
              </li>
              <li>
                <_default
                  dark={true}
                  emphasized={true}
                >
                  VatNumber
                </_default>
                <_default>
                  1
                </_default>
              </li>
            </ul>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h1>
              Uniliver
            </h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Uniliver",
            },
            "ref": null,
            "rendered": "Uniliver",
            "type": "h1",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": false,
              "rules": Array [
                [Function],
                ";",
                "-webkit-tap-highlight-color:",
                [Function],
                ";border:0;box-sizing:border-box;margin:0;outline:",
                [Function],
                ";padding:0;position:relative;",
                ";",
                "
    margin-top: 34px;
    .input-error {
        color: #cc3333;
    }
",
              ],
            },
            "displayName": "Styled(base-components__RootElement)",
            "foldedComponentIds": Array [
              "base-components__RootElement-sc-150tvch-1",
            ],
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <_default
                className="supplier-title"
                dark={true}
                emphasized={true}
              >
                 Supplier details
              </_default>,
              <ul>
                <li>
                  <_default
                    dark={true}
                    emphasized={true}
                  >
                    partnerName
                  </_default>
                  <_default>
                    Uniliver
                  </_default>
                </li>
                <li>
                  <_default
                    dark={true}
                    emphasized={true}
                  >
                    CompanyHouseName
                  </_default>
                  <_default>
                    uni
                  </_default>
                </li>
                <li>
                  <_default
                    dark={true}
                    emphasized={true}
                  >
                    CompanyRegistrstionNumber
                  </_default>
                  <_default>
                    1
                  </_default>
                </li>
                <li>
                  <_default
                    dark={true}
                    emphasized={true}
                  >
                    VatNumber
                  </_default>
                  <_default>
                    1
                  </_default>
                </li>
              </ul>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": " Supplier details",
                "className": "supplier-title",
                "dark": true,
                "emphasized": true,
              },
              "ref": null,
              "rendered": " Supplier details",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  Array [
                    <li>
                      <_default
                        dark={true}
                        emphasized={true}
                      >
                        partnerName
                      </_default>
                      <_default>
                        Uniliver
                      </_default>
                    </li>,
                    <li>
                      <_default
                        dark={true}
                        emphasized={true}
                      >
                        CompanyHouseName
                      </_default>
                      <_default>
                        uni
                      </_default>
                    </li>,
                    <li>
                      <_default
                        dark={true}
                        emphasized={true}
                      >
                        CompanyRegistrstionNumber
                      </_default>
                      <_default>
                        1
                      </_default>
                    </li>,
                    <li>
                      <_default
                        dark={true}
                        emphasized={true}
                      >
                        VatNumber
                      </_default>
                      <_default>
                        1
                      </_default>
                    </li>,
                  ],
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <_default
                        dark={true}
                        emphasized={true}
                      >
                        partnerName
                      </_default>,
                      <_default>
                        Uniliver
                      </_default>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "partnerName",
                        "dark": true,
                        "emphasized": true,
                      },
                      "ref": null,
                      "rendered": "partnerName",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "Uniliver",
                      },
                      "ref": null,
                      "rendered": "Uniliver",
                      "type": [Function],
                    },
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <_default
                        dark={true}
                        emphasized={true}
                      >
                        CompanyHouseName
                      </_default>,
                      <_default>
                        uni
                      </_default>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "CompanyHouseName",
                        "dark": true,
                        "emphasized": true,
                      },
                      "ref": null,
                      "rendered": "CompanyHouseName",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "uni",
                      },
                      "ref": null,
                      "rendered": "uni",
                      "type": [Function],
                    },
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <_default
                        dark={true}
                        emphasized={true}
                      >
                        CompanyRegistrstionNumber
                      </_default>,
                      <_default>
                        1
                      </_default>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "CompanyRegistrstionNumber",
                        "dark": true,
                        "emphasized": true,
                      },
                      "ref": null,
                      "rendered": "CompanyRegistrstionNumber",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "1",
                      },
                      "ref": null,
                      "rendered": "1",
                      "type": [Function],
                    },
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "3",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <_default
                        dark={true}
                        emphasized={true}
                      >
                        VatNumber
                      </_default>,
                      <_default>
                        1
                      </_default>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "VatNumber",
                        "dark": true,
                        "emphasized": true,
                      },
                      "ref": null,
                      "rendered": "VatNumber",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "1",
                      },
                      "ref": null,
                      "rendered": "1",
                      "type": [Function],
                    },
                  ],
                  "type": "li",
                },
              ],
              "type": "ul",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-VigVT",
              "isStatic": true,
              "rules": Array [
                "
    background-color: #f6f6f6;
    padding: 28px 12px 16px 12px;
    border: solid 1px #e5e5e5;
    label.supplier-title{
        font-size: 20px;
        margin-bottom: 20px
    }
    ul{
        margin:0
        padding:0
        li {
            width: 25%;
            display: inline-block;
            margin-bottom: 20px;
            label{
                width: 100%;
            }                  
            @media (max-width: 756px) {
                width: 50%;
            }  
            @media (max-width: 505px) {
                width: 100%;
            }       
        }
    }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-VigVT",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
